bpf/polprog/pol_prog_builder.go

```diff
@@ -302,6 +304,29 @@ func (p *Builder) writeRule(rule *proto.Rule, passLabel string) {
 		p.writePortsMatch(true, legDest, rule.NotDstPorts, rule.NotDstNamedPortIpSetIds)
 	}
 
+	if rule.IpVersion == 4 {
```

Comment:
Don't think you need this check here.  When we start handling ICMPv6 we'll use the same location in the state struct so the match logic will still be correct,.

Reply:
For IPv6,  can we still use Rule_IcmpTypeCode?